### PR TITLE
Admin area to manage extended metadata, and disable various types

### PR DESCRIPTION
**Pull request by stuzart**
A page to admin Extended Metadata, currently just to list and allow them to be disbaled.
Disabling will make them unavailable in the UI, and hide any existing entries from both UI and API. Existing ones can still be edited.

* fix for #1649

### CODE REVIEW

File: app/views/admin/_general.html.erb
Comment (whomingbird) [inline app/views/admin/_general.html.erb:1]: <li><%= link_to 'Manage extended metadata types', administer_extended_metadata_types_path %></li>
should you standardize the case of this “extended metadata” ？

File: app/views/admin/_general.html.erb [hunk 1]
```diff
@@ -33,6 +33,7 @@
         <li><%= link_to 'Manage assay types', suggested_assay_types_path %></li>
         <li><%= link_to 'Manage technology types', suggested_technology_types_path %></li>
         <li><%= link_to 'Manage tags', tags_admin_path %></li>
+        <li><%= link_to 'Manage extended metadata types', administer_extended_metadata_types_path %></li>
```
Comment (whomingbird) [inline app/views/admin/_general.html.erb:36]: should you standardize the case of this “extended metadata” ？

File: app/views/extended_metadata_types/_extended_metadata_type_table_row.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  <td><%= extended_metadata_type.title %></td>
+  <td><%= extended_metadata_type.supported_type %></td>
+  <td><%= extended_metadata_type.extended_metadatas.count %></td>
+  <td><%= extended_metadata_type.enabled? ? 'Enabled' : 'Disabled' %></td>
+  <td>
+    <% unless extended_metadata_type.extended_type? %>
+      <%= form_for(extended_metadata_type, url: administer_update_extended_metadata_type_path(extended_metadata_type), method: :put) do |f| %>
+        <%= f.hidden_field :enabled, value: !extended_metadata_type.enabled? %>
+        <% if extended_metadata_type.enabled? %>
+          <%= f.submit 'Disable', class:'btn' %>
```
Comment (whomingbird) [inline app/views/extended_metadata_types/_extended_metadata_type_table_row.html.erb:12]: maybe could add some color for the buttons

```
     <%= f.submit 'Disable', class:'btn btn-danger' %>
        <% else %>
      <%= f.submit 'Enable', class:'btn btn-success' %>
```

File: app/views/extended_metadata_types/_extended_metadata_type_table_row.html.erb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<tr class="<%= 'warning' if extended_metadata_type.disabled_but_used? %>">
```
Comment (whomingbird) [inline app/views/extended_metadata_types/_extended_metadata_type_table_row.html.erb:1]: maybe also add a css to turn the text color in the row to **grey** when EM is disabled.  ( optional )

**Review by whomingbird — CHANGES_REQUESTED** [review]

File: app/views/admin/_general.html.erb (hunk 1 above)
Comment (stuzart) [reply app/views/admin/_general.html.erb:36]: yeah, I had noticed there is an inconsistency with Workflow Class, I will capitalize them all

image

File: app/views/extended_metadata_types/_extended_metadata_type_table_row.html.erb (hunk 2 above)
Comment (stuzart) [reply app/views/extended_metadata_types/_extended_metadata_type_table_row.html.erb:12]: I did wonder about that, but maybe btn-warning for Disable, danger seems a bit scarey

Comment (whomingbird) [reply app/views/extended_metadata_types/_extended_metadata_type_table_row.html.erb:12]: It would be good just to distinguish two statuses in any way.

**Pull request merged**